CSP sandbox allow-same-origin - clarify (#40489)
mdn/content: files/en-us/web/http/reference/headers/content-security-policy/sandbox/index.md

- `allow-presentation`
  - : Allows embedders to have control over whether an iframe can start a [presentation session](/en-US/docs/Web/API/PresentationRequest).
- `allow-same-origin`
  - : Allows a same-origin sandboxed resource to be treated as same-origin, reducing the restrictions normally applied to a sandbox.

    If this token is not used, the resource is treated as being from a special origin that always fails the {{Glossary("same-origin policy")}} (potentially preventing access to [data storage/cookies](/en-US/docs/Web/Security/Same-origin_policy#cross-origin_data_storage_access) and some JavaScript APIs).

>     The {{httpheader("Origin")}} of such a request would be `null`.

Reviewer: ```suggestion
    The {{httpheader("Origin")}} of sandboxed resources without the `allow-same-origin` keyword is `null`.
```

- `allow-presentation`
  - : Allows embedders to have control over whether an iframe can start a [presentation session](/en-US/docs/Web/API/PresentationRequest).
- `allow-same-origin`
  - : Allows a sandboxed resource to retain its {{Glossary("origin")}}.

>     A sandboxed resource is otherwise treated as being from an opaque origin, which ensures that it will always fail {{Glossary("same-origin policy")}} checks, and hence cannot access [`localstorage` and `document.cookie`](/en-US/docs/Web/Security/Same-origin_policy#cross-origin_data_storage_access) and some JavaScript APIs.

Reviewer: I think we don't define "opaque origin" anywhere, maybe there's a follow-up opportunity on the `origin` glossary entry?

```suggestion
    A sandboxed resource is otherwise treated as being from an _opaque origin_, which ensures that it will always fail {{Glossary("same-origin policy")}} checks, and hence cannot access [`localstorage` and `document.cookie`](/en-US/docs/Web/Security/Same-origin_policy#cross-origin_data_storage_access) and some JavaScript APIs.
```

When searching, I find:

```
./files/en-us/web/api/credentialscontainer/get/index.md:    - The calling origin is an [opaque origin](/en-US/docs/Web/HTTP/Reference/Headers/Origin#null).
```

Any thoughts on linking to https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Origin#null?
